sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/ParallelDocumentQueryExecutionContextBase.java

```diff
@@ -38,7 +38,7 @@
 public abstract class ParallelDocumentQueryExecutionContextBase<T>
         extends DocumentQueryExecutionContextBase<T> implements IDocumentQueryExecutionComponent<T> {
 
-    protected final List<DocumentProducer<T>> documentProducers;
```

Comment:
This change looks very risky - too easy to screw something up (thread-safety etc.) - why do you need to make the List mutable? Looks liek that needs some refactoring

Reply:
I agree this is a risky change. But if we don't refresh the documentProducers, we will end up having duplicate results when running the componentQueries, because after every componentQuery is runs, we doing a Flux.fromIterable on the list of documentProducers and returning all the results.

Reply:
I'm not certain what we can in this scenario, we can discuss this on Monday.